llama_index/indices/base.py

```diff
@@ -189,23 +189,61 @@ def insert(self, document: Document, **insert_kwargs: Any) -> None:
 
     @abstractmethod
     def _delete(self, doc_id: str, **delete_kwargs: Any) -> None:
-        """Delete a document."""
+        """Delete a node."""
+
+    def delete_nodes(
+        self,
+        doc_ids: List[str],
+        delete_from_docstore: bool = False,
+        **delete_kwargs: Any,
+    ) -> None:
+        """Delete a list of nodes from the index.
+
+        Args:
+            doc_ids (List[str]): A list of doc_ids from the nodes to delete
+
+        """
+        for doc_id in doc_ids:
+            self._delete(doc_id, **delete_kwargs)
+            if delete_from_docstore:
+                self.docstore.delete_document(doc_id, raise_error=False)
```

Comment:
curious: should this always be managed by the super class? How can we convey to someone implementing a index subclass to not worry about deleting stuff from the docstore?

Reply:
I think since nodes can be shared between indexes (something we advertise) subclasses shouldn't need to worry about (the base class handles all docstore interaction currently)